### PR TITLE
Fix Java version in README

### PR DESCRIPTION
**Pull request by der-eismann**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ Apache Commons Lang, a package of Java utility classes for the
 
 The code is tested using the latest revision of the JDK for supported
 LTS releases.
-These are 8, 11, 12 and 21 currently, see:
+These are 8, 11, 17 and 21 currently, see:
```
Comment (garydgregory) [inline README.md:59]: This is generated from the `pom.xml` file.

**Comment by der-eismann** [comment]
> * Apache is the canonical repository at [gitbox.apache.org/repos/asf/commons-lang.git](https://gitbox.apache.org/repos/asf/commons-lang.git)
>   The fact that [gitbox.apache.org/repos/asf?p=commons-lang.git](https://gitbox.apache.org/repos/asf?p=commons-lang.git) redirects here is a bug with Apache's infrastructure.

But https://gitbox.apache.org/repos/asf/commons-lang.git redirects to GitHub as well, doesn't it? Actually I just got the idea because I saw it over at  [commons-codec](https://github.com/apache/commons-codec/blob/master/pom.xml#L48) and thought it should be the same everywhere.

File: README.md (hunk 1 above)
Comment (garydgregory) [reply README.md:59]: > Both pom.xml and readme.md were changed in [this commit](https://github.com/apache/commons-lang/commit/9a306b2b7abe3bd38595fa0d9fc3a5e9e35a70d4), so what exactly is wrong?

Hello @der-eismann 
Please do not change the URL.

File: pom.xml [hunk 2]
```diff
@@ -53,7 +53,7 @@
   <scm>
     <connection>scm:git:http://gitbox.apache.org/repos/asf/commons-lang.git</connection>
     <developerConnection>scm:git:https://gitbox.apache.org/repos/asf/commons-lang.git</developerConnection>
-    <url>https://gitbox.apache.org/repos/asf?p=commons-lang.git</url>
+    <url>https://github.com/apache/commons-lang</url>
```
Comment (garydgregory) [inline pom.xml:56]: The Apache repository is the canonical one, the redirect is a bug 😞

**Pull request merged**